### ISSUE TITLE
Protect against the default `.js` extension being replaced before pirates gets a chance to store `originalJsExtension`

### ISSUE THREAD

**Issue by jamestalmage** [ariporad/pirates#24]
Right now `pirates` assumes that `Module_.extensions['.js']` points to the original javascript extension the first time it sees it. That is an improper assumption, as it could have been changed already by any one of the following:
1. Another module that doesn't use pirates.
2. Another module that uses a semver incompatible version of pirates (so it's not deduped).
3. Another module that uses pirates in an `npm@2` environment (so there is no de-duping regardless).

Reference:
- https://github.com/jamestalmage/default-require-extensions. Should provide an easy fix for this.
- https://github.com/jamestalmage/append-transform/issues/6. Reasons we chose this way vs alternative solutions.

**Comment by ariporad** [comment]
Hmm... I'm not 100% sure about this... It might be valid that you want to copy
whatever the chain for .js is. (ie. You're adding a clmpile-to-js file format).
Let me think on it.

Thanks! Ari
On Tue, Jan 26, 2016 at 7:26 PM, James Talmage <notifications@github.com> wrote:
Right now pirates assumes that Module_.extensions['.js'] points to the original javascript extension. That is an improper assumption, it
could have been changed already by any one of the following:

 1. Another module that doesn't use pirates.
 2. Another module that uses a semver incompatible version of pirates (so it's
    not deduped).
 3. Another module that uses pirates in an npm@2 environment (so there is no de-duping regardless).

Reference:

 * [https://github.com/jamestalmage/default-require-extensions] https://github.com/jamestalmage/default-require-extensions
   [https://github.com/jamestalmage/default-require-extensions] . Should provide an easy fix for this.
 * jamestalmage/append-transform#6
   [https://github.com/jamestalmage/append-transform/issues/6] . Reasons we chose this way vs alternative solutions.
